checks/linux.py

```diff
@@ -0,0 +1,176 @@
+            Please note that the availability of captures and specific features depends on your Desktop Environment's implementation of these portals.<br><br>
+            Global Keyboard Shortcuts are not currently available under Wayland."""]
+
+
+def checkX11(lines):
+    isDistroNix = search('Distribution:', lines)
+    isFlatpak = search('Flatpak Runtime:', lines)
+
+    if (len(isDistroNix) <= 0) and (len(isFlatpak) <= 0):
+        return
+
+    sessionTypeLine = getSessionTypeLine(lines)
+    if not sessionTypeLine:
+        return
+
+    sessionType = sessionTypeLine.split()[3]
+    if sessionType != 'x11':
+        return
+
+    # obsolete PW sources
+    hasPipewireCaptureDesktop = search('pipewire-desktop-capture-source', lines)
+    hasPipewireCaptureWindow = search('pipewire-window-capture-source', lines)
+    # unified PW source
+    hasPipewireCaptureScreen = search('pipewire-screen-capture-source', lines)
+
+    if (len(hasPipewireCaptureDesktop) > 0) or (len(hasPipewireCaptureWindow) > 0) or (len(hasPipewireCaptureScreen) > 0):
+        return [LEVEL_WARNING, "PipeWire capture on X11",
+                """While technically possible, most Desktop Environments do not implement the PipeWire capture portals on X11. Your captures will therefore likely not work, i.e. you will be unable to pick a window or display, or the selected source will stay empty.<br><br>
+                We generally recommend using \"Window Capture (Xcomposite)\" on X11, as \"Display Capture (XSHM)\" can introduce bottlenecks depending on your setup.<br><br>
+                From a technical standpoint a Desktop Environment should not advertise the availability of these captures if they are not implemented. This behaviour can therefore be considered a bug in your Desktop Environment."""]
+
+    return [LEVEL_INFO, "X11",
+            "Window Capture is available via Xcomposite, while Display Capture is available via XSHM. We generally recommend sticking to \"Window Capture (Xcomposite)\" since \"Display Capture (XSHM)\" can introduce bottlenecks depending on your setup."]
+
+
+def checkDesktopEnvironment(lines):
+    isDistroNix = search('Distribution:', lines)
+    isFlatpak = search('Flatpak Runtime:', lines)
+
+    if (len(isDistroNix) <= 0) and (len(isFlatpak) <= 0):
+        return
+
+    desktopEnvironmentLine = search('Desktop Environment:', lines)
+    desktopEnvironment = desktopEnvironmentLine[0].split()
+    desktopEnvironment = desktopEnvironment[3:]
+    desktopEnvironment = ' '.join(desktopEnvironment)
+
+    if (len(desktopEnvironment) > 0):
+        return [LEVEL_INFO, desktopEnvironment, '']
+
+
+def checkMissingModules(lines):
+    isDistroNix = search('Distribution:', lines)
+
+    if (len(isDistroNix) <= 0):
+        return
+
+    modulesMissingList = []
+    modulesCheckList = ('obs-browser.so', 'obs-websocket.so', 'vlc-video.so')
```

Comment:
What's the rationale for checking only these modules?

Reply:
I was trying to find a balance between checking "everything" and those that might reasonably be missing and are actually asked for in support. This PR was also opened before the CMake rewrite hit master so a lot of things that might be disabled by default now used to be enabled by default so I figured it wouldn't make much sense to include.

Specifically for this list:
Most distros don't ship Browser by default, many don't ship Websocket due to missing dependencies, and VLC source requires VLC to be installed (and isn't available in Flatpak at all). So those are the 3 major ones that could be missing.
As noted in the OP I left out AJA and BlackMagic because they are virtually never asked for.

It just occurred to me that `obs-x264.so` might be useful to check for, although it's really only relevant for Fedora at the moment (the plugin is shipped in RPM Fusion not in Fedora repos).

That being said, I could check for all modules if desired.

Reply:
I don't have a strong opinion on this. I just wanted to raise a question.